File: Greedy/1975. Maximum Matrix Sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) 
    {
        long long sum = 0;
        int maxValue = INT_MAX, neg = 0;
        for(auto vec:matrix)
        {
            for(auto val:vec)
            {
                sum += abs(val);
                if(val < 0) neg += 1;
                maxValue = min(maxValue, abs(val));
            }
        }

        if(neg % 2) sum -= maxValue * 2;

        return sum;
    }
};

File: Greedy/1975. Maximum Matrix Sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        negetive, positive = [], []
        for arr in matrix:
            for val in arr:
                if val > 0:
                    positive.append(val)
                else:
                    negetive.append(val)
        min_num = 0
        if len(negetive) % 2:
            min_num = max(max(negetive + [-10**5]), -min(positive + [10**5]))
        return sum(positive) + (sum(negetive) * - 1) + (min_num + min_num)